### PR TITLE
refactor: move BrowserIsOutdatedBanner from layout

### PR DESCRIPTION
**Pull request by johangirod**
This PR refactor the BrowserIsOutdatedBanner component from the layouts component into the generic one in _app.tsx

It also changes the message to present what is working instead of what is not.

### TODO 

- investigate why the warning message is displayed for IE 11 and not for firefox < 65 in prod only (it works in local and in preview apps)

### Screenshot

image

### CODE REVIEW

**Comment by XavierJp** [comment]
Really like what you did with the refacto. I wonder if we can even move it out of `postServerSideProps` ?

**Comment by XavierJp** [comment]
Interesting read, but do not seem to take very old browser into account : 
https://andreasremdt.com/blog/detecting-outdated-browsers/

Another idée : use feature detection and triger banner from frontend : 
https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection

Possible links (but private companies) : 
https://browser-update.org/update-browser.html
https://bestvpn.org/outdatedbrowser/en.html

File: components/banner/browser-is-outdated.tsx [hunk 1]
```diff
@@ -32,12 +32,29 @@ export const BrowserIsOutdatedBanner: React.FC<{}> = () => (
...
+      //  https://caniuse.com/?search=esmodule
+      //  https://stackoverflow.com/questions/44891421/detect-es6-import-compatibility
+      dangerouslySetInnerHTML={{
+        __html: `
+          const esModuleDisabled = typeof window !== 'undefined' 
+            && !('noModule' in HTMLScriptElement.prototype)
+            
+          if (esModuleDisabled) {
+            getElementById('browser-is-outdated').style.display = 'block'
+          }
```
Comment (XavierJp) [inline components/banner/browser-is-outdated.tsx:47]: Not sure `const`, `&&` and `style.display` is backward compatible with very old browsers. At least we should encapsulate iin a try catch that would default on displaying the banner

File: package.json [hunk 2]
```diff
@@ -39,7 +39,6 @@
     "@types/qrcode": "^1.5.2",
     "@types/react": "^18.2.28",
     "@types/react-dom": "^18.2.13",
-    "@types/ua-parser-js": "^0.7.37",
```
Comment (XavierJp) [inline package.json:42]: I love when we remove packages <3

**Review by XavierJp — CHANGES_REQUESTED** [review]
I fear this might not work on very old browsers

File: components/banner/browser-is-outdated.tsx (hunk 1 above)
Comment (johangirod) [reply components/banner/browser-is-outdated.tsx:47]: I think we're safe for && and style
- https://caniuse.com/mdn-html_elements_style
- https://caniuse.com/mdn-javascript_operators_logical_and

I can replace const for var to allow IE < 11

File: components/banner/browser-is-outdated.tsx [hunk 3]
```diff
@@ -5,38 +5,65 @@ import constants from '#models/constants';
...
+      //  https://caniuse.com/?search=esmodule
+      //  https://stackoverflow.com/questions/44891421/detect-es6-import-compatibility
+      dangerouslySetInnerHTML={{
+        __html: `
+          var esModuleDisabled = typeof window !== 'undefined' 
+            && !('noModule' in HTMLScriptElement.prototype)
+            
+          if (esModuleDisabled) {
+            getElementById('browser-is-outdated').style.display = 'block'
+          }
```
Comment (XavierJp) [inline components/banner/browser-is-outdated.tsx:47]:
```suggestion
	  try {
             var esModuleDisabled = typeof window !== 'undefined' 
               && !('noModule' in HTMLScriptElement.prototype);
            
             if (esModuleDisabled) {
                 throw new Error('browser is outdated');
          } catch {
             getElementById('browser-is-outdated').style.display = 'block';
          }
```

**Review by XavierJp — APPROVED** [review]

**Pull request merged**